libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/TelemetryWrapper.kt

```diff
@@ -0,0 +1,131 @@
+package com.mapbox.navigation.core.telemetry
+
+import androidx.annotation.UiThread
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.telemetry.CustomEvent
+import com.mapbox.navigation.core.internal.telemetry.UserFeedbackCallback
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent
+import com.mapbox.navigation.core.telemetry.events.FeedbackMetadata
+import com.mapbox.navigation.core.telemetry.events.FeedbackMetadataWrapper
+import com.mapbox.navigation.metrics.MapboxMetricsReporter
+import com.mapbox.navigation.metrics.internal.TelemetryUtilsDelegate
+import com.mapbox.navigation.utils.internal.assertDebug
+import com.mapbox.navigation.utils.internal.logD
+
+/**
+ * Class that manages [MapboxNavigationTelemetry] initialisation. Listens to telemetry state events
+ * and can enable/disable telemetry in runtime.
+ *
+ * TODO(NAVAND-1820) ensure that we have 1-1 relationship between [TelemetryWrapper] and [MapboxNavigationTelemetry]
+ * [MapboxNavigationTelemetry] is very complex already and needs to be refactored.
+ */
+@UiThread
+internal class TelemetryWrapper {
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var navigationOptions: NavigationOptions
+    private lateinit var userAgent: String
+
+    private var isWrapperInitialized = false
+    private var sdkTelemetry: SdkTelemetry = SdkTelemetry.EMPTY
+
+    fun initialize(
+        mapboxNavigation: MapboxNavigation,
+        navigationOptions: NavigationOptions,
+        userAgent: String
+    ) {
+        assertDebug(!isWrapperInitialized) {
+            "Already initialized"
+        }
+
+        if (isWrapperInitialized) {
+            return
+        }
+
+        isWrapperInitialized = true
+
+        this.mapboxNavigation = mapboxNavigation
+        this.navigationOptions = navigationOptions
+        this.userAgent = userAgent
+
+        if (TelemetryUtilsDelegate.getEventsCollectionState()) {
+            initializeSdkTelemetry()
```

Comment:
Hmm, what if someone disables the telemetry at runtime? Should we stop sending events? It's not supported now, right?